### PR TITLE
Doc: Small dev model tweaks

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

File: doc/nrf/ug_dev_model.rst [hunk 1]
```diff
@@ -27,6 +27,23 @@ Basic familiarity with Git is required to understand the architecture of the rep
...
 All |NCS| repositories are publicly hosted on `GitHub`_, and accessible to both individual users and companies.
 
+A note about forks, upstreams and downstreams
+=============================================
+
+Some key concepts used throughout this guide deserve their own introduction to avoid confusion among users not experienced with Git and multiple repositories.
+
+Git is a distributed version control system that allows repositories to be easily duplicated, creating copies of them. Every time you take an existing Git repository and create a copy of it, you are effectively creating a *fork* of that repository. This means that you now have an identical copy that might diverge from the original over time, since commits to the original will not be automatically reflected in the copy.
+
+.. note:
```
Comment (mbolivar) [inline doc/nrf/ug_dev_model.rst]: This is a comment, not a note directive, should be `.. note::`

File: doc/nrf/ug_dev_model.rst [hunk 2]
```diff
@@ -27,6 +27,23 @@ Basic familiarity with Git is required to understand the architecture of the rep
...
+   When we talk about forks or copying Git repositories we refer to the creation
+   of a new repository hosted in a server and accessible to other users. If you
```
Comment (mbolivar) [inline doc/nrf/ug_dev_model.rst]: s/in a server/on a server/

Here and below ("hosted in any server").

File: doc/nrf/ug_dev_model.rst [hunk 3]
```diff
@@ -27,6 +27,23 @@ Basic familiarity with Git is required to understand the architecture of the rep
...
+   are simply cloning a repository to your local machine then that is referred
+   to as a *clone*, and not a fork.
+
+Whenever you create a fork by creating a copy of an existing repository, the original repository can be called the *upstream*, and the newly created copy conversely the *downstream*.
+
+A fork can be hosted in any server, including public Git hosting site such as `GitHub`_. It is however important to differentiate between the generic concept of fork introduced earlier in this section and GitHub's own concept of a `GitHub fork`_.  A GitHub fork is GitHub's own concept of a fork: it includes both creating a copy of the original repo being forked and also tagging the downstream or fork with a flag that allows users to send Pull Requests from the fork to its upstream. GitHub also supports creating forks without using this particular feature, simply by creating an empty repository and then pushing the contents of the repository to be cloned to it manually.
```
Comment (mbolivar) [inline doc/nrf/ug_dev_model.rst]: s/public/a public/

I am not sure pull request is a proper noun phrase in this context. Should be lowercase if not.

Are you sure about the correctness of "GitHub also supports creating forks without using this particular feature, simply by creating an empty repository and then pushing the contents of the repository to be cloned to it manually."?

If you make an empty repository, there is no fork relationship in the GitHub sense AFAIK. Is there some way to manually say "I'm a fork of this other repo over here"? If so, might want to say so in this doc, which only mentions pushing.

Comment (carlescufi) [reply doc/nrf/ug_dev_model.rst]: GitHub allows forks in the common sense of the work, it's documented here:
https://help.github.com/en/articles/duplicating-a-repository
I will rework this section to reflect that.

**Comment by carlescufi** [comment]
@mbolivar comments addressed
@ru-fu could you take a look?

**Comment by ru-fu** [comment]
Can you add the source for the PNG file?

**Comment by carlescufi** [comment]
> This PR is not according to the doc process.
> That being said, this user guide needs some overhaul anyway. So I'll do a quick language review now, and we can clean it up later.
> But please follow the doc process in the future.

Apologies, will do

**Comment by carlescufi** [comment]
> Can you add the source for the PNG file?

added

**Comment by carlescufi** [comment]
> I added a commit.

Squashed.

**Comment by carlescufi** [comment]
Thanks @ru-fu this should be ready now.

**Pull request merged**